algorithms/src/Kokkos_Sort.hpp

```diff
@@ -146,8 +146,12 @@ class BinSort {
         Kokkos::is_view<SrcViewType>::value,
         Kokkos::View<typename SrcViewType::const_data_type,
                      typename SrcViewType::array_layout,
-                     typename SrcViewType::device_type,
-                     Kokkos::MemoryTraits<Kokkos::RandomAccess> >,
+                     typename SrcViewType::device_type
+#if !defined(KOKKOS_COMPILER_NVHPC)  // FIXME_NVHPC
+                     ,
+                     Kokkos::MemoryTraits<Kokkos::RandomAccess>
+#endif
```

Comment:
Is this ifdef guard necessary for unit tests in this PR to work?
I'm curious what caused this ifdef for `NVHPC` to be added, seeing that the ifdef below is still in the unit test.
```
#if defined(KOKKOS_ENABLE_CUDA) && \
    defined(KOKKOS_COMPILER_NVHPC)  // FIXME_NVHPC
  if (!std::is_same_v<ExecutionSpace, Kokkos::Cuda>)
#endif
```
Not asking for any change, just looking for a clarification.

Reply:
I have not modified the existing tests, but i added TestBinSortB which needs that guard to work. Makes sense?